src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/resilience/RollbacksSpec.groovy

```diff
@@ -112,4 +117,140 @@ and at least 1 path must remain safe"
+                it.state == IslChangeType.FAILED
+            }.size() == broughtDownPorts.size() * 2
+        }
+
+        and: "A protected flow"
+        /** At this point we have the following topology:
+         *
+         *   srcSwitch - - - - - dstSwitch <- swToManipulate
+         *          \              /
+         *           \           /
+         *           transitSwitch
+         *
+         **/
+        def flow = flowHelperV2.randomFlow(swPair)
+        flow.allocateProtectedPath = true
+        flowHelperV2.addFlow(flow)
+        def flowPathInfo = northbound.getFlowPath(flow.flowId)
+        assert pathHelper.convert(flowPathInfo) == mainPath
+        assert pathHelper.convert(flowPathInfo.protectedPath) == protectedPath
+
+        and: "Disconnect dst switch transit switch on protected path"
+        def swToManipulate = swPair.dst
+        def blockData = switchHelper.knockoutSwitch(swToManipulate, mgmtFlManager)
+        def isSwitchActivated = false
+
+        when: "Mark the transit switch as ACTIVE in db"
+        database.setSwitchStatus(swToManipulate.dpId, SwitchStatus.ACTIVE)
+
+        and: "Init flow #data.description"
+        data.action(flow)
+
+        then: "System retried to #data.description"
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getFlowHistory(flow.flowId).findAll {
+                it.action == data.historyAction
+            }.last().histories*.details.findAll{ it =~ /.+ Retrying/}.size() == data.historySize
+            println(data.historySize)
+        }
+
+        then: "Flows are not rerouted"
+        def flowPathInfoAfterSwap = northbound.getFlowPath(flow.flowId)
+        assert pathHelper.convert(flowPathInfoAfterSwap) == mainPath
+        assert pathHelper.convert(flowPathInfoAfterSwap.protectedPath) == protectedPath
+
+        and: "Flow is DOWN"
+        Wrappers.wait(WAIT_OFFSET * 2) { // delete 'wait' when 3237 is merged
+            assert northboundV2.getFlowStatus(flow.flowId).status == FlowState.DOWN
+        }
+
+        and: "All involved switches pass switch validation(except dst switch)"
+        def involvedSwitchIds = pathHelper.getInvolvedSwitches(protectedPath)[0..-2]*.dpId
+        Wrappers.wait(WAIT_OFFSET / 2) {
+            involvedSwitchIds.each { swId ->
+                with(northbound.validateSwitch(swId)) { validation ->
+                    validation.verifyRuleSectionsAreEmpty(["missing", "excess", "misconfigured"])
+                    validation.verifyMeterSectionsAreEmpty(["missing", "excess", "misconfigured"])
+                }
+            }
+        }
+
+        when: "Connect dst switch back to the controller"
+        database.setSwitchStatus(swToManipulate.dpId, SwitchStatus.INACTIVE) //set real status
+        switchHelper.reviveSwitch(swPair.dst, blockData)
+        isSwitchActivated = true
+
+        then: "Flow is UP"
+        Wrappers.wait(discoveryInterval + rerouteDelay) {
```

Comment:
pretty tight timeout with no room for actual reroute operations. May be unstable